### PR TITLE
[flutter_tools] remove globals from compile and devices

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Remove global variables from compile.dart and devices.dart (except for dds which needs more changes).

https://github.com/flutter/flutter/issues/47161

### CODE REVIEW

File: packages/flutter_tools/lib/src/device.dart [hunk 1]
```diff
@@ -275,11 +287,11 @@ abstract class DeviceManager {
...
       // choose one. If the user has two nonEphemeral devices running, we also
       // request input to choose one.
-      if (devices.length > 1 && globals.stdio.stdinHasTerminal) {
-        globals.printStatus(globals.userMessages.flutterMultipleDevicesFound);
-        await Device.printDevices(devices);
+      if (devices.length > 1 && _terminal.stdinHasTerminal) {
+        _logger.printStatus(_userMessages.flutterMultipleDevicesFound);
+        await Device.printDevices(devices, _logger);
         final Device chosenDevice = await _chooseOneOfAvailableDevices(devices);
-        globals.deviceManager.specifiedDeviceId = chosenDevice.id;
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/device.dart:282]: huh

File: packages/flutter_tools/test/general.shard/device_test.dart [hunk 2]
```diff
@@ -557,7 +549,6 @@ class TestDeviceManager extends DeviceManager {
   }
 }
 
-class MockProcess extends Mock implements Process {}
```
Comment (jmagman) [inline packages/flutter_tools/test/general.shard/device_test.dart:560]: Woo!

**Review by jmagman — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/device.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/device.dart:282]: 🤷‍♂️

**Pull request merged**